docs/diga_api.md

```diff
@@ -0,0 +1,289 @@
+
+- `EDFC0` and `TDFC0` for verification
+- `EDRE0` and `TDRE0` for billing.
+
+## Nutzdaten
+
+`nutzdaten` is the tricky field which contains verification or billing related information based on the `verfahren` value.
+`nutzdaten` are defined in an xml sheet which needs to be encrypted and encoded as byte array before sending the actual requests.
+The structure of the xml files are defined by `xsd` files which can be found in `src/main/resources` or downloaded from [gkv page](https://www.gkv-datenaustausch.de/leistungserbringer/digitale_gesundheitsanwendungen/digitale_gesundheitsanwendungen.jsp) as [attachment - DiGA-Abrechnungsschema (XSD) (ZIP)](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_XSD_20201029.zip).
+Examples for the data and additional details on the encryption are described in the following.
+
+# Request details
+
+## Code verification requests
+
+See `EDFC0-Pruefung_2.0.0.xsd` file for reference:
+
+Data for sending the request:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<Pruefung_Freischaltcode xmlns="http://www.gkv-datenaustausch.de/XML-Schema/EDFC0_Pruefung/2.0.0"
+                         version="002.000.000" gueltigab="2020-07-01" verfahrenskennung="TDFC0" nachrichtentyp="ANF"
+                         absender="987654321" empfaenger="123456789">
+    <Antwort>
+        <IK_DiGA_Hersteller>123456789</IK_DiGA_Hersteller>
+        <IK_Krankenkasse>987654321</IK_Krankenkasse>
+        <DiGAID>12345000</DiGAVEID>
+        <Freischaltcode>ABCDEFGHIJKLMNOP</Freischaltcode>
+    </Antwort>
+</Pruefung_Freischaltcode>
+```
+
+In case you are not an approved DiGA manufacturer yet, you can put `12345` as `DiGAID`.
+You should already have an `IK` number though as otherwise, you also wouldn't have the certificate for encryption/decryption.
+
+Successful response data:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<Pruefung_Freischaltcode xmlns="http://www.gkv-datenaustausch.de/XML-Schema/EDFC0_Pruefung/2.0.0"
+                         version="002.000.000" gueltigab="2020-07-01" verfahrenskennung="TDFC0" nachrichtentyp="ANT"
+                         absender="987654321" empfaenger="123456789">
+    <Antwort>
+        <IK_DiGA_Hersteller>123456789</IK_DiGA_Hersteller>
+        <IK_Krankenkasse>987654321</IK_Krankenkasse>
+        <DiGAVEID>12345000</DiGAVEID>
+        <Freischaltcode>ABCDEFGHIJKLMNOP</Freischaltcode>
+        <Tag_der_Leistungserbringung>2020-08-19</Tag_der_Leistungserbringung>
+    </Antwort>
+</Pruefung_Freischaltcode>
+```
+
+The response differs from the request by having `nachrichtentyp="ANT"` as header (`Antwort`) instead of `nachrichtentyp="ANF"` (`Anfrage`).
+Also the diga id is listed as `DiGAVEID` (verified ID) and an additional field `Tag_der_Leistungserbringung` is returned (day when the service started).
+
+## Billing requests
```

Comment:
I was a bit shorter on this section as we will first focus on the verification. Also the xml files are more complex I think. Would be great to add some example request/response data here as well.

Reply:
Yes, I also found the one from the test but wanted to use some xml files from the "real" api. But I am still blocked by the IK number and certificates and therefore, cannot send test requests.

Reply:
I got the diga-api-client up and running locally and extracted the post data during debugging. However, I failed to send the requests and got the following traceback. Any idea how to solve?

```
java.net.ProtocolException: expected the Content-Type to have a boundary parameter
        at okhttp3.MultipartReader.<init>(MultipartReader.kt:87)
        at com.alextherapeutics.diga.implementation.DigaOkHttpClient.parseResponse(DigaOkHttpClient.java:79)
        at com.alextherapeutics.diga.implementation.DigaOkHttpClient.post(DigaOkHttpClient.java:70)
        at com.alextherapeutics.diga.DigaApiClient.performCodeValidation(DigaApiClient.java:226)
        at com.alextherapeutics.diga.DigaApiClient.sendTestCodeValidationRequest(DigaApiClient.java:153)
        at com.alextherapeutics.diga.Main.main(Main.java:57)
11:18:43.767 [main] ERROR com.alextherapeutics.diga.DigaApiClient - Failed to validate DiGA code 77AAAAAAAAAAAAAX
com.alextherapeutics.diga.DigaHttpClientException: Diga Http Client error
        at com.alextherapeutics.diga.implementation.DigaOkHttpClient.post(DigaOkHttpClient.java:73)
        at com.alextherapeutics.diga.DigaApiClient.performCodeValidation(DigaApiClient.java:226)
        at com.alextherapeutics.diga.DigaApiClient.sendTestCodeValidationRequest(DigaApiClient.java:153)
        at com.alextherapeutics.diga.Main.main(Main.java:57)
Caused by: java.net.ProtocolException: expected the Content-Type to have a boundary parameter
        at okhttp3.MultipartReader.<init>(MultipartReader.kt:87)
        at com.alextherapeutics.diga.implementation.DigaOkHttpClient.parseResponse(DigaOkHttpClient.java:79)
        at com.alextherapeutics.diga.implementation.DigaOkHttpClient.post(DigaOkHttpClient.java:70)
        ... 3 common frames omitted
```

This also happened when I started the `diga-api-service` locally. Using the service with the provided dockerfile I was able to make requests. This is where I got the response data from.